keps/sig-autoscaling/3148-hpa-dryrun/README.md

```diff
@@ -0,0 +1,865 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+To be able to get the insights on pod scaling mechanisms, one needs to be able to run HPA without actually rescaling pods. In cases when scaling mechanisms (based on in-built/custom metrics) include multiple stages such as testing the scaling decision, validating and approving it (whether the output of scaling mechanism is correct) - being able to dry-run HPA seems benefiacial.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- allow users to run HPA in a dry-run mode
+- allow users to set dry-run mode dynamically
+- allow users to see the inforamtion on proposed scaling solution
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+TBD
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+There should be a way to enable dry-run mode when createing autoscaler, such as:
+
+
+```
+kubectl autoscale deployment php-apache --cpu-percent=50 --dryrunEnable=true
+```
+
+The default `dryrunEnable` should be set to `false`, so the user can enable it if it is needed.
+
+Users should be able to dynamically set `dryrunEnable` on running HPA through editing HPA resource.
+
+A typical example `yaml` file to describe HPA looks like following 
+
+```yaml
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: php-apache
+  minReplicas: 1
+  maxReplicas: 10
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 50
+```
+
+With the addition of `dryrunEnable` flaf, user should be able to enable HPA dryRun mode. The user is not expectede to provide the `minReplicas` and `maxReplicas` in this case 
```

Comment:
Nit: "flag" (spelling)

Reply:
"expected"

Reply:
I would avoid making validation weaker during dry run.  When enabling actuation (turning dry run off) you should exactly what you saw before, including the min/max limits.  Besides, max is not a reference; it's by value so it's always required.